thirtyinch/src/main/java/net/grandcentrix/thirtyinch/internal/PresenterSavior.java

```diff
@@ -52,7 +52,7 @@ public TiPresenter recover(final String id) {
     }
 
     public String safe(@NonNull final TiPresenter presenter) {
-        final String id = generateId(presenter);
+        final String id = presenter.getId();
         TiLog.v(TAG, "safe presenter with id " + id + " " + presenter);
```

Comment:
Because it's the presenter's ID. Otherwise it would be an ID for a presenter used by the `PresenterSavior`. I'm using the ID at other places.